### PR TITLE
feat(Mute): add the feature to mute messages

### PR DESCRIPTION
**Pull request by NextAlone**
Signed-off-by: NextAlone <12210746+NextAlone@users.noreply.github.com>

# feat(Mute): add the feature to mute messages

## 更改内容 / Types of Changes

- 新功能 New Feature

## 由这个PR解决或关闭的Issues / Issues Fixed or Closed by This PR

-  

## 检查列表 / Checklist

- [x] 我的代码符合本仓库的代码规范 My code follows the code style of this project
- [x] 我已经测试了这些更改，它们可以正常工作，既不会破坏原有任何功能(或我可以解决)，也不会影响对旧版本的支持 I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] 我已合并了对后续工作无意义的commit并确认不会对后续维护造成困扰 I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

File: app/src/main/java/cn/lliiooll/activity/MuteCfgActivity.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    @Override
+    public boolean doOnCreate(Bundle bundle) {
+        super.doOnCreate(bundle);
+        LinearLayout ll = new LinearLayout(this);
+        ll.setOrientation(LinearLayout.VERTICAL);
+        ViewGroup.LayoutParams mmlp = new ViewGroup.LayoutParams(MATCH_PARENT, MATCH_PARENT);
+        LinearLayout __ll = new LinearLayout(MuteCfgActivity.this);
+        __ll.setOrientation(LinearLayout.VERTICAL);
+        ViewGroup bounceScrollView = new BounceScrollView(this, null);
+        //invoke_virtual(bounceScrollView,"a",true,500,500,boolean.class,int.class,int.class);
```
Comment (qwq233) [inline app/src/main/java/cn/lliiooll/activity/MuteCfgActivity.java:58]: delete pls

File: app/src/main/java/cn/lliiooll/activity/MuteCfgActivity.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        bounceScrollView.setLayoutParams(mmlp);
+        bounceScrollView.setId(R.id.rootBounceScrollView);
+        ll.setId(R.id.rootMainLayout);
+        bounceScrollView.addView(ll, new ViewGroup.LayoutParams(MATCH_PARENT, WRAP_CONTENT));
+        //invoke_virtual(bounceScrollView,"setNeedHorizontalGesture",true,boolean.class);
```
Comment (qwq233) [inline app/src/main/java/cn/lliiooll/activity/MuteCfgActivity.java:63]: same

File: app/src/main/java/cn/lliiooll/utils/MsgRecord.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
+package cn.lliiooll.utils;
```
Comment (singleNeuron) [inline app/src/main/java/cn/lliiooll/utils/MsgRecord.java:1]: 不要同时存在两个MsgRecord
https://github.com/ferredoxin/QNotified/blob/master/app/src/main/java/me/singleneuron/data/MsgRecordData.kt

**Review by singleNeuron — APPROVED** [review]

File: app/src/main/java/cn/lliiooll/utils/MsgType.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
+package cn.lliiooll.utils;
+
+public class MsgType {
+    // from com.tencent.mobileqq.data.MessageRecord
```
Comment (qwq233) [inline app/src/main/java/cn/lliiooll/utils/MsgType.java:4]: same

File: app/src/main/java/cn/lliiooll/utils/MsgRecord.java (hunk 3 above)
Comment (singleNeuron) [inline app/src/main/java/cn/lliiooll/utils/MsgRecord.java:1]: 不要同时存在两个MsgRecord
https://github.com/ferredoxin/QNotified/blob/master/app/src/main/java/me/singleneuron/data/MsgRecordData.kt

**Review by singleNeuron — CHANGES_REQUESTED** [review]

**Review by qwq233 — CHANGES_REQUESTED** [review]
顺便问一句 这玩意确定能用吗

**Comment by NextAlone** [comment]
> 顺便问一句 这玩意确定能用吗

能。测过了。啊这。

File: app/src/main/java/cn/lliiooll/activity/MuteCfgActivity.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import nil.nadph.qnotified.ui.ResUtils;
+
+import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;
+import static android.view.ViewGroup.LayoutParams.WRAP_CONTENT;
+import static nil.nadph.qnotified.ui.ViewBuilder.newListItemHookSwitchInit;
+import static nil.nadph.qnotified.ui.ViewBuilder.subtitle;
+import static nil.nadph.qnotified.util.Utils.dip2px;
+
+@SuppressLint("Registered")
+public class MuteCfgActivity extends IphoneTitleBarActivityCompat implements View.OnClickListener {
```
Comment (lliioollcn) [inline app/src/main/java/cn/lliiooll/activity/MuteCfgActivity.java:45]: 建议你改改再提交，或者等我优化优化在提交...

**Review by lliioollcn — CHANGES_REQUESTED** [review]

File: app/src/main/java/cn/lliiooll/activity/MuteCfgActivity.java (hunk 5 above)
Comment (NextAlone) [reply app/src/main/java/cn/lliiooll/activity/MuteCfgActivity.java:45]: 那就先挂着吧，这几天也没时间

**Pull request closed without merging**